Ariel/Ariel/Models/DailyWeather.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  Weather.swift
+//  Ariel
+//
+//  Created by Alina Zaitseva on 5/16/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+
+struct DailyWeather {
+    let summary:String
+    let icon:String
+    let temperature:Double
+    
+    enum SerializationError:Error {
+        case missing(String)
+        case invalid(String, Any)
+    }
+    
+    init(json:[String:Any]) throws {
+        guard let summary = json["summary"] as? String else {throw SerializationError.missing("summary is missing")}
+        
+        guard let icon = json["icon"] as? String else {throw SerializationError.missing("icon is missing")}
+        
+        guard let temperature = json["temperatureMax"] as? Double else {throw SerializationError.missing("temp is missing")}
+        
+        self.summary = summary
+        self.icon = icon
+        self.temperature = temperature
+    }
+    
+    static let basePath = "https://api.darksky.net/forecast/2c73b2906faf137f2f19c38fc3454588/"
+    static func forecast (withLocation location:CLLocationCoordinate2D, completion: @escaping ([DailyWeather]?) -> ()) {
+        
+        let url = basePath + "\(location.latitude),\(location.longitude)"
+        let request = URLRequest(url: URL(string: url)!)
+        
+        let task = URLSession.shared.dataTask(with: request) { (data:Data?, response:URLResponse?, error:Error?) in
+            
+            var forecastArray:[DailyWeather] = []
```

Comment:
You're always returning array - so no necessity in optional in completion closure